From: Jiri Kosina <jkosina@suse.cz>
Patch-mainline: No, SUSE specific
References: bsc#1092497
Subject: [PATCH] x86/bugs: correctly force-disable IBRS on !SKL systems

spectre_v2_select_mitigation() intends to force-disable IBRS in case the
selection logic picks (any form of) retpoline as suitable mitigation, and
we're not running on SKL+ CPU.

There is no point on making this decision on the value of ibrs_state at that
point; we want to set it to '0' (force disabled), and later let
x86_spec_check() make the has-it-been-force-disabled decision based on it.

Fixes: 12d4b3c51bc2 ("x86/retpolines/spec_ctrl: disable IBRS on !SKL if retpolines are active")
Signed-off-by: Jiri Kosina <jkosina@suse.cz>
---
 arch/x86/kernel/cpu/bugs.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/arch/x86/kernel/cpu/bugs.c
+++ b/arch/x86/kernel/cpu/bugs.c
@@ -412,7 +412,7 @@ retpoline_auto:
 		pr_info("Filling RSB on context switch\n");
 	}
 
-	if (!is_skylake_era() && x86_ibrs_enabled()) {
+	if (!is_skylake_era()) {
 		pr_info("Retpolines enabled, force-disabling IBRS due to !SKL-era core\n");
 		ibrs_state = 0;
 	}
